hakrawler.go

```diff
@@ -233,17 +235,27 @@ func extractHostname(urlString string) (string, error) {
 }
 
 // print result constructs output lines and sends them to the results chan
-func printResult(link string, sourceName string, showSource bool, showJson bool, results chan string, e *colly.HTMLElement) {
+func printResult(link string, sourceName string, showSource bool, showWhere bool, showJson bool, results chan string, e *colly.HTMLElement) {
 	result := e.Request.AbsoluteURL(link)
+	whereURL := e.Request.URL.String()
 	if result != "" {
 		if showJson {
+		    where := ""
+		    if showWhere {
+			    where = whereURL
+			}
 			bytes, _ := json.Marshal(Result{
 				Source: sourceName,
 				URL:    result,
+				Where:  where,
 			})
 			result = string(bytes)
 		} else if showSource {
-			result = "[" + sourceName + "] " + result
+		    whereResult := ""
+			if showWhere {
+			    whereResult = "[" + whereURL + "] "
+			}
+			result = whereResult + "[" + sourceName + "] " + result
```

Comment:
This implementation requires the `-s` flag to be set for the `-w` flag to output where the path was found.
```
# this shows where the URL is found
echo "https://www.google.com" | ./hakrawler -d 2 -w -s
# but this does not
echo "https://www.google.com" | ./hakrawler -d 2 -w
```

What if you leaved the showSource condition as is, and add this as a condition below the show source condition:
```
if !showJson && showWhere {
	result = "[" + whereURL + "] " + result
}
```

This will allow the user to set the `-w` flag without setting the `-s` flag.

It might also make sense to use different characters to denote the where the path is found, since brackets are already used by the output of the `-s` flag. Maybe `whereResult = "<" + whereURL + "> "` ?

I'm not a maintainer of this project. I think this feature you are adding is useful and want to see this merged.

Reply:
Hi Erik,

Thank you for reviewing. I made the fix as you said. Currently, it works without any flag dependency.

> It might also make sense to use different characters to denote the where the path is found, since brackets are already used by the output of the -s flag. Maybe whereResult = "<" + whereURL + "> " ?

For this, I wanted to develop in accordance with the existing structure.